compiler/il/Aliases.cpp

```diff
@@ -797,6 +797,10 @@ OMR::SymbolReference::setSharedShadowAliases(TR_BitVector * aliases, TR::SymbolR
       while (symRef = i.getNext())
          if (symRef->getSymbol() == self()->getSymbol())
             aliases->set(symRef->getReferenceNumber());
+
+      // include symbol reference's own shared alias bitvector
```

Comment:
Okay, I see your point about setting the flag being something expected at sym ref creation time currently (though it seems like a convention that could be relaxed if we wanted since it's not very pervasive in the codebase). My point was : why not do everything needed to make a sym ref an alias of the other if the makeSharedAliases api was called ?  Having said that, at least for union sym refs (the use case you are going after here) one should know that sharing is possible at sym ref creation time. So, maybe an assert is good enough for now to raise the red flag if the api was used without "really shares symbol" being set in some future scenario.